contracts/strategies/lift/LiftStrategy.sol

```diff
@@ -0,0 +1,272 @@
+    if (IERC20(underlying).balanceOf(address(this)) > 0) {
+      IERC20(underlying).safeTransfer(vault, IERC20(underlying).balanceOf(address(this)));
+    }
+  }
+
+  /*
+  *   Withdraws all the asset to the vault
+  */
+  function withdrawToVault(uint256 amount) public restricted {
+    // Typically there wouldn't be any amount here
+    // however, it is possible because of the emergencyExit
+    if(amount > IERC20(underlying).balanceOf(address(this))){
+      // While we have the check above, we still using SafeMath below
+      // for the peace of mind (in case something gets changed in between)
+      uint256 needToWithdraw = amount.sub(IERC20(underlying).balanceOf(address(this)));
+      rewardPool.withdraw(Math.min(rewardPool.balanceOf(address(this)), needToWithdraw));
+    }
+    IERC20(underlying).safeTransfer(vault, amount);
+  }
+
+  /*
+  *   Note that we currently do not have a mechanism here to include the
+  *   amount of reward that is accrued.
+  */
+  function investedUnderlyingBalance() external view returns (uint256) {
+    if (address(rewardPool) == address(0)) {
+      return IERC20(underlying).balanceOf(address(this));
+    }
+    // Adding the amount locked in the reward pool and the amount that is somehow in this contract
+    // both are in the units of "underlying"
+    // The second part is needed because there is the emergency exit mechanism
+    // which would break the assumption that all the funds are always inside of the reward pool
+    return rewardPool.balanceOf(address(this)).add(IERC20(underlying).balanceOf(address(this)));
+  }
+
+  /*
+  *   Governance or Controller can claim coins that are somehow transferred into the contract
+  *   Note that they cannot come in take away coins that are used and defined in the strategy itself
+  *   Those are protected by the "unsalvagableTokens". To check, see where those are being flagged.
+  */
+  function salvage(address recipient, address token, uint256 amount) external onlyControllerOrGovernance {
+     // To make sure that governance cannot come in and take away the coins
+    require(!unsalvagableTokens[token], "token is defined as not salvagable");
+    IERC20(token).safeTransfer(recipient, amount);
+  }
+
+  /*
+  *   Get the reward, sell it in exchange for underlying, invest what you got.
+  *   It's not much, but it's honest work.
+  *
+  *   Note that although `onlyNotPausedInvesting` is not added here,
+  *   calling `investAllUnderlying()` affectively blocks the usage of `doHardWork`
+  *   when the investing is being paused by governance.
+  */
+  function doHardWork() external onlyNotPausedInvesting restricted {
+    uint256 rewardBalanceBefore = boardRoom.getbalanceOfShare(address(this));
+    rewardPool.stakeInBoardroom();
+    uint256 rewardBalanceAfter = boardRoom.getbalanceOfShare(address(this));
+    uint256 rewardStaked = rewardBalanceAfter.sub(rewardBalanceBefore);
+    if (rewardStaked > 0) {
+      stakes.push([block.timestamp, rewardStaked]);
```

Comment:
Is there any issue if two or more stakes have the same `block.timestamp`? (I could see it happen if txs get stuck in the pending state and then all get mined in the same block)